jujugui/static/gui/src/app/components/login/login.js

```diff
@@ -23,14 +23,28 @@ YUI.add('login-component', function() {
   juju.components.Login = React.createClass({
 
     propTypes: {
+      controllerAPI: React.PropTypes.object.isRequired,
       errorMessage: React.PropTypes.string,
+      gisf: React.PropTypes.bool.isRequired,
       isLegacyJuju: React.PropTypes.bool.isRequired,
+      localStorageGet: React.PropTypes.func,
       loginToAPIs: React.PropTypes.func.isRequired,
-      loginToController: React.PropTypes.func.isRequired
+      loginToController: React.PropTypes.func.isRequired,
+      sendPost: React.PropTypes.func
     },
 
     componentDidMount: function () {
       this.refs.username.focus();
+      if (this.props.gisf) {
+        const bounce = () => {
+          if (this.props.controllerAPI.get('connected')) {
```

Comment:
At this point, rather than passing the whole controller API, what do you think about just passing a controllerIsConnected function returning a boolean? This way we stick with the pattern we have about passing verbs to components. In the future, we want to group all those methods in a interface.
Something like:
```javascript
<window.juju.components.Login
  ...
  controllerIsConnected={function() {return controllerAPI.get('connected');}}
  ...
/>
```